### PR TITLE
LIMS-2391: Use source analysis's sample ID in duplicate analysis IDs

### PR DESCRIPTION
**Pull request by rockfruit**
@xispa any reason we used the samplepartition ID originally?

### CODE REVIEW

**Comment by xispa** [comment]
Have no idea... I think using SampleID instead is better. Maybe I'm wrong, but I have the premonition that this change will require mode modifs in the code than only those in the commit d8283c8 . Did all tests passed in your dev environment?

**Comment by rockfruit** [comment]
passed tests and working in the lab.

**Comment by rockfruit** [comment]
I also did a search for 'D' in all code and I can't find any place where there be a problem.

**Comment by xispa** [comment]
> passed tests and working in the lab.

Awesome. Nothing more to say rather accept the PR ;)

File: bika/lims/content/worksheet.py [hunk 1]
```diff
@@ -329,17 +328,16 @@ def addDuplicateAnalyses(self, src_slot, dest_slot):
...
             # the same Reference Sample and same Worksheet)
-            # https://github.com/bikalabs/Bika-LIMS/issues/931
             if not refgid and not analysis.portal_type == 'ReferenceAnalysis':
-                part = analysis.getSamplePartition().id
+                prefix = analysis.aq_parent.getSample().id
                 dups = [an.getReferenceAnalysesGroupID()
                         for an in self.getAnalyses()
                         if an.portal_type == 'DuplicateAnalysis'
-                            and an.getSamplePartition().id == part]
+                            and an.getSamplePartition().id == prefix]
```
Comment (xispa) [inline bika/lims/content/worksheet.py]: umm... wait, don't understand how this line can work unless ```getSamplePartition()``` returns the Sample, which in any case would be wrong.

Comment (xispa) [inline bika/lims/content/worksheet.py]: In fact, if you try to add more than one duplicate for the same analysis, I expect that both duplicates will have the sample ReferenceAnalysesGroupID, which shouldn't be

Comment (rockfruit) [inline bika/lims/content/worksheet.py]: Hum, you are right.  How did I miss it?

**Pull request merged**